static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.ts

```diff
@@ -0,0 +1,169 @@
+import raven from '../../../../lib/raven';
+import { removeSlots } from '../../../commercial/modules/remove-slots';
+import { getPageTargeting } from '../../../common/modules/commercial/build-page-targeting';
+import { commercialFeatures } from '../../../common/modules/commercial/commercial-features';
+import type { IdentityUser } from '../../../common/modules/identity/api';
+import { getUserFromApi } from '../../../common/modules/identity/api';
+import { adFreeSlotRemove } from '../ad-free-slot-remove';
+import { init as initMessenger } from '../messenger';
+import { init as background } from '../messenger/background';
+import { init as sendClick } from '../messenger/click';
+import { init as disableRefresh } from '../messenger/disable-refresh';
+import { init as initGetPageTargeting } from '../messenger/get-page-targeting';
+import { init as initGetPageUrl } from '../messenger/get-page-url';
+import { init as getStyles } from '../messenger/get-stylesheet';
+import { init as hide } from '../messenger/hide';
+import { init as resize } from '../messenger/resize';
+import { init as scroll } from '../messenger/scroll';
+import { init as type } from '../messenger/type';
+import { init as viewport } from '../messenger/viewport';
+import { fillAdvertSlots } from './fill-advert-slots';
+import { onSlotLoad } from './on-slot-load';
+import { onSlotRender } from './on-slot-render';
+import { onSlotViewableFunction } from './on-slot-viewable';
+import { onSlotVisibilityChanged } from './on-slot-visibility-changed';
+import { refreshOnResize } from './refresh-on-resize';
+
+initMessenger(
+	type,
+	getStyles,
+	initGetPageTargeting,
+	initGetPageUrl,
+	initMeasureAdLoad,
+	resize,
+	hide,
+	scroll,
+	viewport,
+	sendClick,
+	background,
+	disableRefresh,
+);
+
+const setDfpListeners = (): void => {
+	const pubads = window.googletag?.pubads();
+	if (!pubads) return;
+	pubads.addEventListener(
+		'slotRenderEnded',
+		raven.wrap(onSlotRender) as (arg0: any) => void,
+	);
+	pubads.addEventListener(
+		'slotOnload',
+		raven.wrap(onSlotLoad) as (arg0: any) => void,
+	);
+
+	pubads.addEventListener('impressionViewable', onSlotViewableFunction());
+
+	pubads.addEventListener('slotVisibilityChanged', onSlotVisibilityChanged);
+	if (storage.session.isAvailable()) {
+		const pageViews =
+			(storage.session.get('gu.commercial.pageViews') as number) || 0;
+		storage.session.set('gu.commercial.pageViews', pageViews + 1);
+	}
+};
+
+const setPageTargeting = (): void => {
+	const pubads = window.googletag?.pubads();
+	if (!pubads) return;
+	// because commercialFeatures may export itself as {} in the event of an exception during construction
+	const targeting = getPageTargeting() as Record<string, any>;
+	Object.keys(targeting).forEach((key) => {
+		pubads.setTargeting(key, targeting[key]);
+	});
+};
+
+const setPublisherProvidedId = (): void => {
+	// Also known as PPID
+	getUserFromApi((user: IdentityUser) => {
+		if (user.privateFields.googleTagId) {
```

Comment:
This should check that the user exists, and that this field is set, otherwise we get errors in the console.

`user?.privateFields?.googleTagId`